testplan/runnable/base.py

```diff
@@ -945,7 +945,8 @@ def _post_exporters(self):
                 report_urls.append(exporter.report_url)
             if getattr(exporter, "_web_server_thread", None):
                 # Give priority to open report from local server
-                webbrowser.open(exporter.report_url)
+                if self.cfg.browse:
+                    webbrowser.open(exporter.report_url)
                 report_opened = True
```

Comment:
I know with the current implementation it does not make a difference but `report_opened = True` should only be set when browser opened.

Reply:
Done